libp2p/protocols/pubsub/gossipsub/behavior.nim

```diff
@@ -310,18 +310,21 @@ proc rebalanceMesh*(g: GossipSub, topic: string, metrics: ptr MeshMetrics = nil)
   if npeers  < g.parameters.dLow:
     trace "replenishing mesh", peers = npeers
     # replenish the mesh if we're below Dlo
-    var candidates = toSeq(
-      g.gossipsub.getOrDefault(topic, initHashSet[PubSubPeer]()) -
-      g.mesh.getOrDefault(topic, initHashSet[PubSubPeer]())
-    ).filterIt(
-      it.connected and
-      # avoid negative score peers
-      it.score >= 0.0 and
-      # don't pick explicit peers
-      it.peerId notin g.parameters.directPeers and
-      # and avoid peers we are backing off
-      it.peerId notin g.backingOff.getOrDefault(topic)
-    )
+    var candidates: seq[PubSubPeer]
+    let currentMesh = g.mesh.getOrDefault(topic)
+    g.gossipSub.withValue(topic, peerList):
+      for it in peerList[]:
+        if
+          it notin currentMesh and
+          it.connected and
+          # avoid negative score peers
+          it.score >= 0.0 and
+          # don't pick explicit peers
+          it.peerId notin g.parameters.directPeers and
+          # and avoid peers we are backing off
+          it.peerId notin g.backingOff.getOrDefault(topic):
+
+          candidates.add(it)
```

Comment:
as a general thing, I like when there's a 4-space indent of the if-stuff so that the 2-space indent of the code gets differentiated, indent-wise